docs/coding.md

```diff
@@ -39,26 +39,58 @@ Note that `IdString`s need a `Context` (or `BaseCtx`) pointer to convert them ba
 ## Developing CAD algorithms - packing
 
 Packing in nextpnr could be done in two ways (if significant packing is done at all):
- - replacing multiple cells with a single larger cell that corresponds to a bel
+ - replacing multiple cells with a single larger cell that corresponds to a BEL
  - combining smaller cells using relative placement constraints
 
-In flows with minimal packing the main task of the packer is to transform cells into common types that correspond to a bel (e.g. convert multiple flipflop primitives to a single common type with some extra parameters). The packer will also have to add relative constraints for fixed structures such as carry chains or LUT-MUX cascades.
+The packer will also have to add relative constraints for fixed structures such as carry chains or LUT-MUX cascades.
 
 There are several helper functions that are useful for developing packers and other passes that perform significant netlist modifications in `util.h`. It is often preferable to use these compared to direct modification of the netlist structures due to the "double linking" nextpnr does - e.g. when connecting a port you must update both the `net` field of the ports and the `driver`/`users` of the net.
 
+### Cell to BEL mapping
+
+There is an Arch API choice when it comes to representing the relationship
+between cell types and BEL types.  One option is to transform cells into
+common types that correspond to a BEL (e.g. convert multiple flipflop
+primitives to a single common type with some extra parameters). In Arch APIs
+designed like this, packer transformations are required to convert input cell
+types into nextpnr specific cell types that have a 1 to 1 relationship with
+BEL types.
+
+For Arch APIs of this type, the method `isValidBelForCellType` reduces to:
+
+```
+bool isValidBelForCellType(IdString cell_type, BelId bel) const {
+    return cell_type == getBelType(bel);
+}
+```
+
+The alternative is to implement a fast `isValidBelForCellType` method that
+determines if this cell type can be bound to this BEL.
+
 ## Developing CAD algorithms - placement
 
-The job of the placer in nextpnr is to find a suitable bel for each cell in the design; while respecting legality and relative constraints.
+The job of the placer in nextpnr is to find a suitable BEL for each cell in the design; while respecting legality and relative constraints.
```

Comment:
bit of a nitpick but the general convention in nextpnr has not been to have bel/pip in all-caps

Reply:
Sure.